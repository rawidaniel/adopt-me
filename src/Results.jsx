import Pet from "./Pet";

const Results = ({ pets, isLoading }) => {
  // console.log(isLoading);
  return (
    <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 lg:grid-cols-3">
      {isLoading && <h2>Please wait while isLoading.....</h2>}
      {!isLoading && !pets.length ? (
        <h1>No Pets Found</h1>
      ) : (
        pets.map((pet) => (
          <Pet
            animal={pet.animal}
            id={pet.id}
            name={pet.name}
            breed={pet.breed}
            images={pet.images}
            location={`${pet.city}, ${pet.state}`}
            key={pet.id}
          />
        ))
      )}
    </div>
  );
};

export default Results;
