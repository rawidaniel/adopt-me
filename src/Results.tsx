import Pet from "./Pet";
import { Pet as PetType } from "./ApiResponsesTypes";

const Results = ({
  pets,
  isLoading,
}: {
  pets: PetType[];
  isLoading: boolean;
}) => {
  // console.log(isLoading);
  return (
    <div className="search">
      {isLoading && <h2>Please wait while isLoading.....</h2>}
      {!isLoading && !pets.length ? (
        <h1>No Pets Found</h1>
      ) : (
        pets.map((pet) => (
          <Pet
            animal={pet.animal}
            id={pet.id}
            name={pet.name}
            breed={pet.breed}
            images={pet.images}
            location={`${pet.city}, ${pet.state}`}
            key={pet.id}
          />
        ))
      )}
    </div>
  );
};

export default Results;
