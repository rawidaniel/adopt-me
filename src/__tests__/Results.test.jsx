import { test, expect } from "vitest";
import { render } from "@testing-library/react";
import Results from "../Results";
import { StaticRouter } from "react-router-dom/server";

const pets = [
  {
    id: 1,
    name: "Luna",
    animal: "dog",
    city: "Seattle",
    state: "WA",
    description:
      "Luna is actually the most adorable dog in the world. Her hobbies include yelling at squirrels, aggressively napping on her owners' laps, and asking to be fed two hours before IT'S DAMN WELL TIME LUNA. Luna is beloved by her puppy parents and lazily resides currently in Seattle, Washington.",
    breed: "Havanese",
    images: [
      "http://pets-images.dev-apis.com/pets/dog25.jpg",
      "http://pets-images.dev-apis.com/pets/dog26.jpg",
      "http://pets-images.dev-apis.com/pets/dog27.jpg",
      "http://pets-images.dev-apis.com/pets/dog28.jpg",
      "http://pets-images.dev-apis.com/pets/dog29.jpg",
    ],
  },
  {
    id: 2,
    name: "Charisse",
    animal: "rabbit",
    city: "Lexington",
    state: "KY",
    description:
      "Maecenas leo odio, condimentum id, luctus nec, molestie sed, justo. Pellentesque viverra pede ac diam. Cras pellentesque volutpat dui.\n\nMaecenas tristique, est et tempus semper, est quam pharetra magna, ac consequat metus sapien ut nunc. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Mauris viverra diam vitae quam. Suspendisse potenti.",
    breed: "Havanese",
    images: ["http://pets-images.dev-apis.com/pets/rabbit0.jpg"],
  },
  {
    id: 3,
    name: "Maitilde",
    animal: "rabbit",
    city: "Dallas",
    state: "TX",
    description:
      "Duis consequat dui nec nisi volutpat eleifend. Donec ut dolor. Morbi vel lectus in quam fringilla rhoncus.\n\nMauris enim leo, rhoncus sed, vestibulum sit amet, cursus id, turpis. Integer aliquet, massa id lobortis convallis, tortor risus dapibus augue, vel accumsan tellus nisi eu orci. Mauris lacinia sapien quis libero.\n\nNullam sit amet turpis elementum ligula vehicula consequat. Morbi a ipsum. Integer a nibh.",
    breed: "Lab",
    images: ["http://pets-images.dev-apis.com/pets/rabbit1.jpg"],
  },
];

test("renders correctly with no pets", async () => {
  const { asFragment } = render(<Results pets={[]} isLoading={false} />);
  expect(asFragment()).toMatchSnapshot();
});

test("renders correctly with pets", async () => {
  const { asFragment } = render(
    <StaticRouter>
      <Results pets={pets} isLoading={true} />
    </StaticRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});
