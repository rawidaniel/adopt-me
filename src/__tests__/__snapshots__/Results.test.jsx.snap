// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly with no pets 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <h1>
      No Pets Found
    </h1>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with pets 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <h2>
      Please wait while isLoading.....
    </h2>
    <a
      class="pet"
      href="/details/1"
    >
      <div
        class="image-container"
      >
        <img
          alt="Luna"
          data-testid="thumbnail"
          src="http://pets-images.dev-apis.com/pets/dog25.jpg"
        />
      </div>
      <div
        class="info"
      >
        <h1>
          Luna
        </h1>
        <h2>
          dog - Havanese - Seattle, WA
        </h2>
      </div>
    </a>
    <a
      class="pet"
      href="/details/2"
    >
      <div
        class="image-container"
      >
        <img
          alt="Charisse"
          data-testid="thumbnail"
          src="http://pets-images.dev-apis.com/pets/rabbit0.jpg"
        />
      </div>
      <div
        class="info"
      >
        <h1>
          Charisse
        </h1>
        <h2>
          rabbit - Havanese - Lexington, KY
        </h2>
      </div>
    </a>
    <a
      class="pet"
      href="/details/3"
    >
      <div
        class="image-container"
      >
        <img
          alt="Maitilde"
          data-testid="thumbnail"
          src="http://pets-images.dev-apis.com/pets/rabbit1.jpg"
        />
      </div>
      <div
        class="info"
      >
        <h1>
          Maitilde
        </h1>
        <h2>
          rabbit - Lab - Dallas, TX
        </h2>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
